wayland-display-core/src/comp/mod.rs

```diff
@@ -476,27 +487,28 @@ pub(crate) fn init(
             if let Err(err) = data
                 .display
                 .handle()
-                .insert_client(client_stream, std::sync::Arc::new(ClientState))
+                .insert_client(client_stream, Arc::new(ClientState::default()))
             {
                 tracing::error!(?err, "Error adding wayland client.");
             };
         })
         .expect("Failed to init wayland socket source");
 
-    event_loop
-        .handle()
-        .insert_source(
-            Generic::new(
-                display.backend().poll_fd().as_raw_fd(),
-                Interest::READ,
-                Mode::Level,
-            ),
-            |_, _, data| {
-                data.display.dispatch_clients(&mut data.state).unwrap();
-                Ok(PostAction::Continue)
-            },
-        )
-        .expect("Failed to init wayland server source");
+    let display_fd = todo!("display.backend().poll_fd().as_raw_fd() -> the trait `AsFd` is not implemented for `i32`");
```

Comment:
I've tried a few variations using `as_fd()` but it errors out

Reply:
Just staff the whole display into `Generic::new`. Also I don't think we need the `Data` struct anymore and can just use `State` directly. We can call `flush_clients` now on a `DisplayHandle`, which we already have in `State`.